pkg/metrics/metrics.go

```diff
@@ -35,6 +35,8 @@ func StartMetrics(config metricsConfig) {
 	http.Handle(config.metricsPath, prometheus.Handler())
 	log.Info("Port: %v", config.metricsPort)
 	metricsPort := ":" + (config.metricsPort)
+	config.withRoute := true
```

Comment:
These should not always be true. Instead,  create two builder functions, `WithRoute()` and `WithServiceMonitor()` that set these.